File: frontend/src/models/analyzer/KeyDto.ts
import type { Kind } from '@/models/analyzer/value/Kind'
import type { RawZonedDateTime } from '@/models/analyzer/common'

type ParentDto = number | null

export interface KeyDto {
  parent: ParentDto
  from: RawZonedDateTime
  to?: RawZonedDateTime
  name: string
  kind: Kind
}

File: analyzer/src/main/java/com/mategka/dava/analyzer/serialization/model/KeyDto.java
package com.mategka.dava.analyzer.serialization.model;

import com.mategka.dava.analyzer.struct.property.value.Kind;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Builder;
import lombok.Data;
import lombok.NonNull;
import org.jetbrains.annotations.Nullable;

import java.time.ZonedDateTime;

@Data
@Builder
public final class KeyDto {

  @NonNull
  final ParentDto parent;
  @NonNull
  final ZonedDateTime from;
  @NonNull
  final String name;
  @NonNull
  final Kind kind;
  @Nullable
  @JsonInclude(JsonInclude.Include.NON_NULL)
  ZonedDateTime to;

}
